account_invoice_merge/models/account_invoice.py

```diff
@@ -59,10 +59,12 @@ def _merge_invoice_line_values(self, vals, new_invoice_line):
         :param new_invoice_line: Recordset of the new line to merge.
         :return: None
         """
+        uom_factor = self.env['product.uom'].browse(
```

Comment:
This will fail if `vals['uos_id']` is incorrect or doesn't exist. You have to do:
```
uom = self.env['product.uom'].browse(vals.get('uos_id', False))
uom_factor = uom.factor if uom.exists() else 1.0
    
```